packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -233,391 +233,616 @@ export class PluginLedgerConnectorIroha
+              key: req.params[1],
+              writer: req.params[2],
+              pageSize: req.params[3],
+              paginationKey: req.params[4],
+              paginationWriter: req.params[5],
+            };
+          } else {
+            params = req.params;
+          }
+          const queryRes = await queries.getAccountDetail(queryOptions, params);
+          return { transactionReceipt: queryRes };
         } catch (err) {
           throw new RuntimeError(err as any);
         }
       }
-      case IrohaQuery.GetEngineReceipts: {
+      case IrohaQuery.GetAssetInfo: {
         try {
-          const response = await queries.getEngineReceipts(queryOptions, {
-            txHash: req.params[0],
-          });
-          return { transactionReceipt: response };
+          let params;
+          if (Array.isArray(req.params)) {
+            params = {
+              assetId: req.params[0],
+            };
+          } else {
+            params = req.params;
+          }
+          const queryRes = await queries.getAssetInfo(queryOptions, params);
+          return { transactionReceipt: queryRes };
         } catch (err) {
           throw new RuntimeError(err as any);
         }
       }
-      case IrohaQuery.FetchCommits: {
+      case IrohaQuery.GetRoles: {
         try {
-          const response = await queries.fetchCommits(queryOptions);
+          const response = await queries.getRoles(queryOptions);
           return { transactionReceipt: response };
         } catch (err) {
           throw new RuntimeError(err as any);
         }
       }
-      case IrohaCommand.AddPeer: {
+      case IrohaQuery.GetRolePermissions: {
         try {
-          const response = await commands.addPeer(commandOptions, {
-            address: req.params[0],
-            peerKey: req.params[1],
-          });
+          let params;
+          if (Array.isArray(req.params)) {
+            params = {
+              roleId: req.params[0],
+            };
+          } else {
+            params = req.params;
+          }
+          const response = await queries.getRolePermissions(
+            queryOptions,
+            params,
+          );
           return { transactionReceipt: response };
         } catch (err) {
           throw new RuntimeError(err as any);
         }
       }
-      case IrohaCommand.RemovePeer: {
+      case IrohaQuery.GetPeers: {
         try {
-          const response = await commands.removePeer(commandOptions, {
-            publicKey: req.params[0],
-          });
+          const response = await queries.getPeers(queryOptions);
           return { transactionReceipt: response };
         } catch (err) {
           throw new RuntimeError(err as any);
         }
       }
-      case IrohaQuery.GetPeers: {
+      case IrohaQuery.FetchCommits: {
         try {
-          const response = await queries.getPeers(queryOptions);
+          const response = await queries.fetchCommits(queryOptions);
```

Comment:
I'm not expert, but one thing looks for me suspiciously: what would happen when not all params would be provided? Or in different order?

E.g.:    
![obraz](https://user-images.githubusercontent.com/15332594/186206178-cf136550-30ae-4218-a805-5dfb80f96980.png)

Reply:
![image](https://user-images.githubusercontent.com/66632190/186419530-eeb8e402-04bf-4150-8171-fb979fbf4a2b.png)
Exactly that's the problem which this PR is going to fix as you can see in the else block we accept the request params in any order. The reason for still keeping the hardcoded way of params is backward compatibility. It would be removed in the future releases. (As discussed with @petermetz)

Reply:
@yashrajdesai @baziorek Yup, the inflexibility of the array based approach is what we are fixing here, but in a backward compatible way. We can drop the array based approach with the release of 2.0 and I already asked @yashrajdesai to create an issue in the tracker specifically for that so for now this thread can be resolved once that GH issue has been opened.

Reply:
@petermetz @baziorek Opened issue [#2151](https://github.com/hyperledger/cactus/issues/2151)  for the same.